tests/canonicalize.rs

```diff
@@ -116,5 +153,9 @@ fn canonicalize_edge_cases() {
         tmpdir.canonicalize("foo/../../"),
         "a path led outside of the filesystem"
     );
-    error_contains!(tmpdir.canonicalize("foo/bar/qux"), "No such file");
+
+    #[cfg(not(windows))]
+    error!(tmpdir.canonicalize("foo/bar/qux"), "No such file");
+    #[cfg(windows)]
+    error!(tmpdir.canonicalize("foo/bar/qux"), 2);
```

Comment:
This is a common enough error that we might defined an `error_no_such_file!` macro or so, so that the tests themselves don't need to have lots of conditional code.

Reply:
This is the approach favoured by libstd hence why I reused it here. But I guess we could tweak provided you’re ok with tweaking libstd-origin tests as well?

Reply:
@sunfishcode Would it make sense to address this (if at all) in a subsequent PR? This is a wider issue than just this PR IMHO since it covers whether we stick with the `libstd` tests or we tweak them.

Reply:
A subsequent PR is ok.

In theory the `libstd` tests all work on Windows already. They have some conditional code to do this, and I think we can mostly leave them as-is, and hopefully they work here too. In our own tests, I'd like to keep an eye out for ways we can factor this kind of thing out, because platform-conditional code makes it much less convenient to write and maintain the tests.

Reply:
OK, gotcha! I'll make sure to address this in a subsequent PR then! :-)